Bind Identity to S3SecurityController

pranavr12 opened the pull request.

mosiac1 reviewed (commented):
LGTM. 

I would to add at least one test that binds an `CredentialsProvider` that will return a non-empty identify and check that the identity reaches `S3SecurityFacadeProvider#securityFacadeForRequest`. We should also bind a custom Identity class. I don't think we currently have any tests that combine all these functionalities.

Updating `TestS3SecurityController` with this might make sense. @vagaerg WDYT?

vagaerg commented:
> LGTM.
> 
> I would to add at least one test that binds an `CredentialsProvider` that will return a non-empty identify and check that the identity reaches `S3SecurityFacadeProvider#securityFacadeForRequest`. We should also bind a custom Identity class. I don't think we currently have any tests that combine all these functionalities.
> 
> Updating `TestS3SecurityController` with this might make sense. @vagaerg WDYT?

Yup that makes sense.

I was also going to suggest waiting for @Randgalt to have a look given this is an SPI change

Randgalt commented on trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/security/S3SecurityFacadeProvider.java:
```diff
@@ -13,17 +13,20 @@
      * throw {@link jakarta.ws.rs.WebApplicationException}
      */
-    S3SecurityFacade securityFacadeForRequest(ParsedS3Request request)
+    S3SecurityFacade securityFacadeForRequest(ParsedS3Request request, Optional<Identity> identity)
```
wdyt about passing the full `Credentials` object instead of just the identity? That might be more flexible for the future. Thoughts?

pranavr12 reviewed (commented):
Currently, the TestingCredentials - https://github.com/trinodb/aws-proxy/blob/main/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingUtil.java#L57 is used by the s3 client. should we add Identity to this ?
If not, how should we plan to test this or should we rely on other tests that the identity from credentials provider is passed to OPA correctly?
Another workaround - In the test class, we could modify the stored credentials in the TestingCredentialsProvider to fetch and restore new credential with identity

pranavr12 commented:
> Currently, the TestingCredentials - https://github.com/trinodb/aws-proxy/blob/main/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingUtil.java#L57 is used by the s3 client. should we add Identity to this ? If not, how should we plan to test this or should we rely on other tests that the identity from credentials provider is passed to OPA correctly? Another workaround - In the test class, we could modify the stored credentials in the TestingCredentialsProvider to fetch and restore new credential with identity

Added the Identoty to TestingCredentials. I have also added an assert in the TestOpaSecurity class to verify the identity
